ChemistryLib/PhreeqcIO.cpp

```diff
@@ -174,6 +209,32 @@ void PhreeqcIO::setAqueousSolutionsOrUpdateProcessSolutions(
     }
 }
 
+void PhreeqcIO::setAqueousSolutionsPrevFromDumpFile()
+{
+    if (!_dump)
+    {
+        return;
+    }
+
+    auto const& dump_file = _dump->dump_file;
+    std::ifstream in(dump_file);
+    if (!in)
+    {
+        OGS_FATAL("Could not open phreeqc dump file '%s'.", dump_file.c_str());
+    }
+
+    std::size_t const num_chemical_systems = _aqueous_solutions.size();
+    _dump->readDumpFile(in, num_chemical_systems);
+
+    if (!in)
+    {
+        OGS_FATAL("Error when reading phreeqc dump file '%s'",
+                  dump_file.c_str());
+    }
+
+    in.close();
```

Comment:
not needed. All objects will be destroyed at closing brace/scope.

Reply:
I prefer to keep it if harmless...